Assets/ConsentManagementProvider/Plugins/iOS/Source/SwiftBridge.swift

```diff
@@ -0,0 +1,334 @@
+        config.vendors = []
+        config.categories = []
+        config.legIntCategories = []
+    }
+    
+    @objc public func dispose() {
+        callbackDefault = nil
+        callbackOnConsentReady = nil
+        callbackOnConsentUIReady = nil
+        callbackOnConsentAction = nil
+        callbackOnConsentUIFinished = nil
+        callbackOnErrorCallback = nil
+        callbackOnSPFinished = nil
+        callbackOnSPUIFinished = nil
+        callbackOnCustomConsent = nil
+    }
+ 
+// MARK: - Manage lib
+    @objc public func loadMessage(authId: String? = nil) {
+        print("PURE SWIFT loadMessage")
+        consentManager.loadMessage(forAuthId: authId)
+    }
+
+    @objc public func onClearConsentTap() {
+        SPConsentManager.clearAllData()
+        myVendorAccepted = .Unknown
+    }
+    
+    @objc public func onGDPRPrivacyManagerTap() {
+        consentManager.loadGDPRPrivacyManager(withId: config.gdprPmId!)
+    }
+    
+    @objc public func onCCPAPrivacyManagerTap() {
+        consentManager.loadCCPAPrivacyManager(withId: config.ccpaPmId!)
+    }
+
+
+    @objc public func customConsentToGDPR() {
+        consentManager.customConsentGDPR(
+            vendors: config.vendors,
+            categories: config.categories,
+            legIntCategories: config.legIntCategories){contents in
+                self.print(contents)
+                self.runCallback(callback: self.callbackOnCustomConsent, arg: contents.toJSON())
+            }
+    }
+
+    @objc public func deleteCustomConsentGDPR() {
+        consentManager.deleteCustomConsentGDPR(
+            vendors: config.vendors,
+            categories: config.categories,
+            legIntCategories: config.legIntCategories){contents in
+                self.print(contents)
+                self.runCallback(callback: self.callbackOnCustomConsent, arg: contents.toJSON())
+            }
+    }
+}
+    
+// MARK: - SPDelegate implementation
+extension SwiftBridge: SPDelegate {
+    public func onSPUIReady(_ controller: UIViewController) {
+        let top = UIApplication.shared.firstKeyWindow?.rootViewController
+        controller.modalPresentationStyle = UIModalPresentationStyle.overFullScreen
+        top?.present(controller, animated: true)
+        logger.log("PURE SWIFT onSPUIReady")
+        runCallback(callback: callbackOnConsentUIReady, arg: "onSPUIReady")
+    }
+    
+    public func onAction(_ action: SPAction, from controller: UIViewController) {
+        print("onAction:", action)
+        logger.log("PURE SWIFT onAction")
+        let responce: Dictionary<String, String> = [
+            "type":String(action.type.rawValue),
+            "customActionId":action.customActionId ?? ""
+        ]
+        var resp = "type:0"
+        if let data = try? JSONEncoder().encode(responce) {
+            resp = String(data: data, encoding: .utf8) ?? ""
+        }
+        runCallback(callback: callbackOnConsentAction, arg: resp)
```

Comment:
I see we're sending `resp = "type:0"` if for some reason we can't encode the "response". Could we send either `nil` or `""` instead?